gtsfm/utils/prepare_nerfstudio.py

```diff
@@ -0,0 +1,162 @@
+"""Functions to create transforms.json file and resized images that are input to Nerfstudio
+
+Here I have modified code taken from Nerfstudio for parsing data in the Nerfstudio format.
+Original files at:
+https://github.com/nerfstudio-project/nerfstudio/blob/main/scripts/process_data.py
+https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py
+
+Author: Jon Womack
+"""
+import argparse
+import json
+import os
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+import thirdparty.nerfstudio.colmap_utils as colmap_utils
+
+
+from gtsfm.utils import images, io
+from gtsam import Rot3
+
+
+class CameraModel(Enum):
+    """Enum for camera types."""
+
+    OPENCV = "OPENCV"
+    OPENCV_FISHEYE = "OPENCV_FISHEYE"
+
+
+CAMERA_MODELS = {
+    "perspective": CameraModel.OPENCV,
+    "fisheye": CameraModel.OPENCV_FISHEYE,
+}
+
+
+def colmap_to_json(cameras_path: Path, images_path: Path, output_dir: Path, camera_model: CameraModel) -> int:
```

Comment:
is this code borrowed form nerfstudio? a pointer would be nice!

Reply:
https://github.com/nerfstudio-project/nerfstudio/blob/1066f8f4330f69363e24984ddf8dc2e7299a23ff/nerfstudio/process_data/colmap_utils.py#L382